src/controllers/hid/hidcontroller.cpp

```diff
@@ -241,23 +209,14 @@ int HidController::open() {
         return -1;
     }
 
+    // Set hid controller to non-blocking
+    if (hid_set_nonblocking(m_pHidDevice, 1) == -1) {
+        qWarning() << "Unable to set HID device " << getName() << " to non-blocking";
```

Comment:
If we are unable to switch into nonblocking mode open() must fail. Otherwise poll() might block forever!

Reply:
Check for != 0 and abort, just to be safe and to reject all unexpected or unknown result codes.